service/storage_service.cc

```diff
@@ -4921,6 +4926,18 @@ future<raft_topology_cmd_result> storage_service::raft_topology_cmd_handler(shar
            // Return an error since the command is from outdated leader
            co_return result;
         }
+
+        {
+            auto& state = _raft_topology_cmd_handler_state;
+            if (state.term != term) {
+                state.term = term;
+            } else if (cmd_index <= state.last_index) {
+                // Return an error since the command is outdated
```

Comment:
Actually we never send uncommitted commands. For committed commands the term never changes. So it should be sufficient to track cmd_index, this will guarantee we never apply the command twice or from an outdated coordinator. What concerns me is that we can miss gaps in cmd_index, e..g due to a software bug we will skip some commands on the follower without noticing. next/prev state id approach guarantees this doesn't happen. but this is outside the scope of this patch.

Reply:
This has nothing to do with raft commands. This is about RPCs from the coordinator.

Reply:
OK, I realized this is a volatile state. OK, so we have raft commit index, state id and now a volatile cmd index which happens to have the same name as raft command index but stands for something different. I hope you see my point: the coordinator state is uniquely identified by raft commit index. It could have been used instead.

Reply:
It cannot. First of all I do not think we should expose it, but more important is that for each raft index we may have multiple rpcs, so it is not unique.